.github/workflows/CI-tag-orchestrator.yml

```diff
@@ -5,54 +5,132 @@ on:
-          DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+      - name: Verify release build
+        run: |
+          ls target
+          ls target/release
+
+      - name: Build solo and relay chain configured images
+        id: build_images
+        run: |
+          SKIP_IMAGE_VALIDATION=true
+          "${GITHUB_WORKSPACE}/docker/scripts/bootstrap.sh" --skip-parachain --skip-compile
+        shell: bash
+
+      - name: Set Docker image names as output
+        id: save-image-names
         run: |
-          source "${GITHUB_WORKSPACE}/ci/setup_env.sh"
-          "${GITHUB_WORKSPACE}/ci/docker.sh" --image-artifact zkverify-image
+          echo "docker_image_solo=horizenlabs/zkverify:latest" >> $GITHUB_OUTPUT
+          echo "docker_image_relay=horizenlabs/zkv-relay:latest" >> $GITHUB_OUTPUT
+
+      - name: Save the Docker image as a tarball
+        id: create-tar-file
+        run: |           
+          TAR_FILE_NAME_SOLO="zkverify-solo"
+          TAR_FILE_SOLO="${{github.workspace}}/${TAR_FILE_NAME_SOLO}.tar"
+
+          TAR_FILE_NAME_RELAY="zkverify-relay"
+          TAR_FILE_RELAY="${{github.workspace}}/${TAR_FILE_NAME_RELAY}.tar"
+
+          echo "tar_file_name_solo=$TAR_FILE_NAME_SOLO" >> $GITHUB_OUTPUT
+          echo "tar_file_solo=$TAR_FILE_SOLO" >> $GITHUB_OUTPUT
+
+          echo "tar_file_name_relay=$TAR_FILE_NAME_RELAY" >> $GITHUB_OUTPUT
+          echo "tar_file_relay=$TAR_FILE_RELAY" >> $GITHUB_OUTPUT
+
+          docker save "${{ steps.save-image-names.outputs.docker_image_solo }}" -o "${TAR_FILE_SOLO}"
+          docker save "${{ steps.save-image-names.outputs.docker_image_relay }}" -o "${TAR_FILE_RELAY}"
+        
+      - name: Upload Docker Image Artifact (solo chain)
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.create-tar-file.outputs.tar_file_name_solo }}
+          path: ${{ steps.create-tar-file.outputs.tar_file_solo }}
+          if-no-files-found: error
+          retention-days: 1
+          compression-level: 0
+          overwrite: true
 
-      - name: Upload runtime artifact
+      - name: Upload Docker Image Artifact (relay chain)
         uses: actions/upload-artifact@v4
         with:
-          name: zkv_runtime.compact.compressed.wasm
-          path: ./zkv_runtime.compact.compressed.wasm
+          name: ${{ steps.create-tar-file.outputs.tar_file_name_relay }}
+          path: ${{ steps.create-tar-file.outputs.tar_file_relay }}
+          if-no-files-found: error
+          retention-days: 1
+          compression-level: 0
+          overwrite: true
+
+  test-publish:
+    needs: build
+    strategy:
+      matrix: 
+        artifact_name:
+        - ${{ needs.build.outputs.artifact_name_solo }}
+        - ${{ needs.build.outputs.artifact_name_relay }}
+    
+    uses: ./.github/workflows/CI-test-publish.yml
+    with:
+      artifact_name: ${{ matrix.artifact_name }}
+      git_tag: ${{ github.event.inputs.git_tag }}
```

Comment:
What if this job was not triggered manually but rather by PUSH ON TAGS? Will this line be still happy :thinking: 
```
github.event.inputs.git_tag
```

Reply:
Good question, it seems to be fine. I triggered this via tag push and didn't get errors on this line

https://github.com/zkVerify/zkVerify/actions/runs/11800656546

Reply:
I think best if this logic is used when the input is defined